hs_core/tasks.py

```diff
@@ -203,6 +207,37 @@ def add_zip_file_contents_to_resource(pk, zip_file_path):
         os.unlink(zip_file_path)
 
 
+@shared_task
+def delete_zip(resource_id, zip_path):
+    from hs_core.hydroshare.utils import get_resource_by_shortkey
+    res = get_resource_by_shortkey(resource_id)
+    istorage = res.get_irods_storage()
+
+    irods_prefix = "/" + settings.IRODS_ZONE + "/home/" + settings.IRODS_USERNAME
+    full_zip_path = '{irods}/{path}'.format(irods=irods_prefix, path=zip_path)
+    istorage.delete(full_zip_path)
+
+
+@shared_task
+def create_temp_zip(resource_id, input_path, output_path):
+    from hs_core.hydroshare.utils import get_resource_by_shortkey
+    res = get_resource_by_shortkey(resource_id)
+    istorage = res.get_irods_storage()
+
+    irods_dest_prefix = "/" + settings.IRODS_ZONE + "/home/" + settings.IRODS_USERNAME
+    full_input_path = '{irods}/{res_id}/{path}'.format(irods=irods_dest_prefix, res_id=resource_id,
+                                                       path=input_path)
+    if res.resource_federation_path:
+        full_input_path = os.path.join(res.resource_federation_path, resource_id, input_path)

```

Comment:
@sblack-usu Looks like you are storing temp zip files in the same iRODS zone as the target HydroShare resource. You can replace lines 227 to 231 with ```full_input_path = os.path.join(res.root_path, input_path)``` which will take federation path into account for you.